server/src/routes/v1/apps/handlers/createApp.js

```diff
@@ -165,6 +165,46 @@ module.exports = {
             const requestUserId = currentUserId
             const developerUserId = appDeveloper.id
 
+            if (appJsonPayload.owner) {
+                const { email, name } = appJsonPayload.owner
+                let appOwner = await getUserByEmail(email, trx)
+
+                if (appOwner === null) {
+                    appOwner = await createUser(
+                        {
+                            email,
+                            name,
+                        },
+                        knex,
+                        trx
+                    )
+                    await addUserToOrganisation(
```

Comment:
this pose a security risk as you could add whatever user you want to any organisation and hijack that organisation as long as the user did not already exist (e.g. create a new user by passing a new e-mail in the owner object pointing to an existing organisation)

Reply:
I agree, good catch! I will look into this.

Reply:
I think https://github.com/dhis2/app-hub/pull/316/commits/ea6ce0c78e7b03bb337190e8db751287708e21ad will fix this. Please check